src/librustc_errors/emitter.rs

```diff
@@ -1200,8 +1200,12 @@ fn num_overlap(a_start: usize, a_end: usize, b_start: usize, b_end:usize, inclus
     } else {
         0
     };
-    (b_start..b_end + extra).contains(a_start) ||
-    (a_start..a_end + extra).contains(b_start)
+    return range_contains(b_start..b_end + extra, a_start) ||
```

Comment:
Would it not be simpler just to have the following? Maybe with a comment for clarity.
```rust
(b_start <= a_start && a_start < b_end + extra) ||
(a_start <= b_start && b_start < a_end + extra);
```